src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -309,7 +310,10 @@ public bool Remove(T item)
                             _slots[last].next = _slots[i].next;
                         }
                         _slots[i].hashCode = -1;
-                        _slots[i].value = default(T);
+                        if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
```

Comment:
Isn't it possible to assign the result of this check to a bool outside for loop, or is the call that cheap?